ext/DifferentiationInterfaceTapirExt/mutating.jl

```diff
@@ -0,0 +1,10 @@
+function DI.value_and_pullback!!(f!, y, dx, ::AutoTapir, x, dy, extras::Nothing)
+    rrule = build_rrule(f!, y, x)
+    dy_righttype = convert(typeof(y), dy)
+    dx_righttype = zero_sametype!!(dx, x)
+    dz = nothing  # f!(y, x) = nothing
```

Comment:
by convention, mutating functions in DI must return z = f!(y, x) = nothing, is this the right way to tell Tapir it should disregard the output cotangent?

Reply:
The tangent type for `nothing` is `NoTangent`, so the only valid thing that could be passed as `dz` is `NoTangent`, so the output cotangent should in effect be disregarded by default provided that `f!` returns `nothing`.

Reply:
```suggestion
    dz = NoTangent()  # f!(y, x) = nothing
```